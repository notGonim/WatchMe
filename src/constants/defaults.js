export { default as Jumbotron } from '../components/jumbotron'
export { default as Footer } from '../components/footer'
export { default as According } from '../components/according'
export { default as Form } from '../components/form'
export { default as Header } from '../components/header'
export { default as Feature } from '../components/feature'

