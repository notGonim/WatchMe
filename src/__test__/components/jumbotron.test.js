import React from 'react';
import { render } from '@testing-library/react';
import jumboData from '../../fixtures/jumbo.json';
import { Jumbotron } from '../../constants/defaults';

describe('<Jumbotron />', () => {
    it('renders the <Jumbotron /> with populated data', () => {
      const { container, getByText, getByAltText, getByTestId } = render(
        <Jumbotron.Container>
          {jumboData.map((item) => (
            <Jumbotron key={item.id}>
              <Jumbotron.Pane>
                <Jumbotron.Title>{item.title}</Jumbotron.Title>
              </Jumbotron.Pane>
              <Jumbotron.Pane>
                <Jumbotron.Image src={item.image} alt={item.alt} data-testid={`${item.id}-jumbo-image`} />
              </Jumbotron.Pane>
            </Jumbotron>
          ))}
        </Jumbotron.Container>
      );
  
      expect(getByText('Enjoy on your TV.')).toBeTruthy();
      expect(getByAltText('Tiger King on Netflix')).toBeTruthy();
      expect(getByTestId('1-jumbo-image')).toBeTruthy();
     
      expect(container.firstChild).toMatchSnapshot();
    });
  });