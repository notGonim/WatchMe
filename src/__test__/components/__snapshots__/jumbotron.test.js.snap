// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders the <Jumbotron /> with populated data 1`] = `
<div
  class="jumbotron__Container-sc-11xlrz1-2 kkpQeT"
>
  <div
    class="jumbotron__Item-sc-11xlrz1-1 kMfAsP"
  >
    <div
      class="jumbotron__Inner-sc-11xlrz1-0 bMklKP"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-11xlrz1-3 fkHQoV"
      >
        <h1
          class="jumbotron__Title-sc-11xlrz1-5 kKzZWR"
        >
          Enjoy on your TV.
        </h1>
      </div>
      <div
        class="jumbotron__Pane-sc-11xlrz1-3 fkHQoV"
      >
        <img
          alt="Tiger King on Netflix"
          class="jumbotron__Image-sc-11xlrz1-6 dxXErT"
          data-testid="1-jumbo-image"
          src="/images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-11xlrz1-1 kMfAsP"
  >
    <div
      class="jumbotron__Inner-sc-11xlrz1-0 bMklKP"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-11xlrz1-3 fkHQoV"
      >
        <h1
          class="jumbotron__Title-sc-11xlrz1-5 kKzZWR"
        >
          Download your programmes to watch on the go.
        </h1>
      </div>
      <div
        class="jumbotron__Pane-sc-11xlrz1-3 fkHQoV"
      >
        <img
          alt="Watch on mobile"
          class="jumbotron__Image-sc-11xlrz1-6 dxXErT"
          data-testid="2-jumbo-image"
          src="/images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-11xlrz1-1 kMfAsP"
  >
    <div
      class="jumbotron__Inner-sc-11xlrz1-0 bMklKP"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-11xlrz1-3 fkHQoV"
      >
        <h1
          class="jumbotron__Title-sc-11xlrz1-5 kKzZWR"
        >
          Watch everywhere.
        </h1>
      </div>
      <div
        class="jumbotron__Pane-sc-11xlrz1-3 fkHQoV"
      >
        <img
          alt="Money Heist on Netflix"
          class="jumbotron__Image-sc-11xlrz1-6 dxXErT"
          data-testid="3-jumbo-image"
          src="/images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
